cmd/ttn-lw-cli/internal/api/grpc.go

```diff
@@ -188,3 +191,62 @@ func CloseAll() {
 		conn.Close()
 	}
 }
+
+// Retry defines retry logic for gRPC calls.
+type Retry[T proto.Message] struct {
+	backoff    time.Duration
+	maxRetries int
+}
+
+// RetryOption defines a functional option for configuring retry behavior.
+type RetryOption[T proto.Message] func(*Retry[T])
+
+// WithBackoff sets the initial backoff duration between retries.
+func WithBackoff[T proto.Message](d time.Duration) RetryOption[T] {
+	return func(r *Retry[T]) {
+		r.backoff = d
+	}
+}
+
+// WithMaxRetries sets the maximum number of retry attempts.
+func WithMaxRetries[T proto.Message](n int) RetryOption[T] {
+	return func(r *Retry[T]) {
+		r.maxRetries = n
+	}
+}
+
+// CallWithBackoff executes a gRPC call with retry logic and exponential backoff.
+//
+// It takes a function `call` that performs the actual gRPC request and retries it
+// if the error returned is a gRPC `codes.ResourceExhausted` error. The retry behavior
+// (initial backoff duration and maximum number of retries) can be customized by
+// providing functional options (e.g., WithBackoff, WithMaxRetries). If no options are
+// provided, it defaults to a 1-second backoff and 5 maximum retries.
+func CallWithBackoff[T proto.Message](call func() (T, error), opts ...RetryOption[T]) (T, error) {
+	r := Retry[T]{
+		backoff:    time.Second,
+		maxRetries: 5,
+	}
+	for _, opt := range opts {
+		opt(&r)
+	}
+	return r.callWithBackoff(call)
+}
```

Comment:
What might be useful to the user is a log informing that there is a timeout going on. 
Maybe pass the context as a parameter and fetch the logger from the context, this way the users won't think there is an error during the timeout.